### PR TITLE
Resolution to Issue 114

### PR DESCRIPTION
**Pull request by Klangina**
# Description

## What kind of change(s) are included?

- [ ] Feature (adds or updates new capabilities)
- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read and followed the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] I have searched for existing content to ensure this is not a duplicate.
- [x] I have performed a self-review of these additions (including spelling, grammar, and related).
- [x] I have added comments to my code to help provide understanding.
- [x] I have added a test which covers the code changes found within this PR.
- [x] I have deleted all non-relevant text in this pull request template.
- [x] **Reviewer assignment**: Tag a relevant team member to review and approve the changes.

### CODE REVIEW

**Comment by Klangina** [comment]
@the-mayer : Kindly review this PR.

File: R/cleanup.R [hunk 1]
```diff
@@ -206,36 +207,23 @@ condenseRepeatedDomains <- function(prot, by_column = "DomArch", excluded_prots
...
+                "\\." = "_d_",
+                " " = "_",
+                "\\+" = " ",
+                "-" = "__",
+                regex_identify_repeats = "\\1(s)",
+                "__" = "-",
+                " " = "+",
+                "_d_" = "."
+            )
+        ))
```
Comment (the-mayer) [inline R/cleanup.R:223]: Very nice -- thank you for streamlining this code, while also fixing the R-CMD check error.

**Review by the-mayer — APPROVED** [review]
Thank you for your contribution @Klangina. I appreciate the additional work to streamline the codebase while also fixing the errors reported by R-CMD check.

File: R/cleanup.R [hunk 2]
```diff
@@ -713,37 +701,37 @@ cleanGeneDescription <- function(prot, column) {
 #' selectLongestDuplicate()
 #' }
 selectLongestDuplicate <- function(prot, column) {
-    col <- sym(column)
-
-    prot$row.orig <- 1:nrow(prot)
-
+    col <- rlang::sym(column)
+    prot <- prot %>%
```
Comment (Joiejoie1) [inline R/cleanup.R:705]: Code Structure & Functionality
Parameter Handling:

The column parameter is converted to a symbol using rlang::sym, enabling flexible selection of any column by name. This is useful for adaptability across datasets with varying column names.

File: R/cleanup.R [hunk 3]
```diff
@@ -713,37 +701,37 @@ cleanGeneDescription <- function(prot, column) {
 #' selectLongestDuplicate()
 #' }
 selectLongestDuplicate <- function(prot, column) {
-    col <- sym(column)
-
-    prot$row.orig <- 1:nrow(prot)
-
+    col <- rlang::sym(column)
+    prot <- prot %>% 
+        mutate(row.orig = seq_len(n()))
```
Comment (Joiejoie1) [inline R/cleanup.R:706]: Marking Original Rows:

mutate(row.orig = seq_len(n())) is used to create a unique identifier for each row, ensuring original row positions are preserved even after filtering. This addition is necessary for accurately managing duplicate entries.

File: R/cleanup.R [hunk 4]
```diff
@@ -713,37 +701,37 @@ cleanGeneDescription <- function(prot, column) {
...
+        group_by(.data$AccNum) %>%
+        summarize(count = n()) %>%
         filter(count > 1) %>%
-        arrange(-count) %>%
-        merge(prot, by = "AccNum")
+        arrange(desc(count)) %>%
+        left_join(prot, by = "AccNum")
 
-    dup_acc <- dups$AccNum
+    dup_acc <- unique(dups$AccNum)
```
Comment (Joiejoie1) [inline R/cleanup.R:715]: Duplicate Identification:

Duplicate entries are identified by grouping on AccNum and then filtered to include only groups with more than one entry. This is well-suited for cases where AccNum uniquely identifies duplicates.

File: R/cleanup.R [hunk 5]
```diff
@@ -713,37 +701,37 @@ cleanGeneDescription <- function(prot, column) {
...
-    longest_rows <- c()
-    remove_rows <- c()
+    longest_rows <- integer()
+    remove_rows <- integer()
     for (acc in dup_acc) {
-        dup_rows <- dups %>% filter(AccNum == acc)
+        dup_rows <- dups %>% filter(.data$AccNum == acc)
 
-        longest <- dup_rows[which(nchar(pull(dup_rows, {{ col }})) == max(nchar(pull(dup_rows, {{ col }}))))[1], "row.orig"]
+        longest <- dup_rows$row.orig[which.max(nchar(pull(dup_rows, !!col)))]
```
Comment (Joiejoie1) [inline R/cleanup.R:722]: Selecting the Longest Entry:

- >For each duplicate group, entries are compared based on the length of text in the specified column (column). The function identifies the row with the longest text entry using which.max(nchar(pull(...))), an efficient approach for selecting the desired row within each duplicate group.
- >Rows not selected as the longest entries are stored in remove_rows to exclude them later.

File: R/cleanup.R [hunk 6]
```diff
@@ -713,37 +701,37 @@ cleanGeneDescription <- function(prot, column) {
...
 
-        # dup_rows[which(nchar(pull(dup_rows,{{col}})) == max(nchar(pull(dup_rows,{{col}}))))[2:nrow(dup_rows)], "row.orig"]
         remove_rows <- c(remove_rows, to_remove)
     }
 
     # grab all the longest rows
-    unique_dups <- prot[-remove_rows, ] %>% select(-row.orig)
+    unique_dups <- prot %>% 
+        filter(!.data$row.orig %in% remove_rows) %>% 
+        select(-.data$row.orig)
```
Comment (Joiejoie1) [inline R/cleanup.R:734]: Filtering and Output:

Finally, filter(!.data$row.orig %in% remove_rows) is used to retain only the longest entries, and row.orig is removed to clean up the output data.

**Review by Joiejoie1 — COMMENTED** [review]
### Recommendations

1. **Code Robustness**:
   - *Plot Function (`plotEstimatedWallTimes`)*: Add checks to ensure `df_walltimes` has the expected structure before performing operations like `gather` and `mutate`. This will help avoid runtime errors or unexpected warnings.
   - *Duplicate Selection Function (`selectLongestDuplicate`)*: Include validation to confirm that the specified `column` exists in `prot`, preventing errors if an invalid column name is provided.

2. **Documentation**:
   - *Plot Function*: Improve documentation by detailing the required structure of `df_walltimes`, helping users understand the data prerequisites.
   - *Duplicate Selection Function*: Add information on the expected structure of `prot`, including that it should contain `AccNum` as a primary identifier, to aid users in correctly preparing their data.

### Summary

Both functions are structured effectively for their purposes, offering clear data transformation, targeted output (a line plot and filtered data frame, respectively), and suitable error handling. Minor documentation enhancements and a few additional error checks would further increase their reliability and usability for developers unfamiliar with the data structure.

**Pull request merged**